converter/src/main/java/gov/cms/qpp/conversion/model/error/Detail.java

```diff
@@ -1,27 +1,59 @@
 package gov.cms.qpp.conversion.model.error;
 
+import java.io.Serializable;
+import java.util.Objects;
+
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
 
-import java.io.Serializable;
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.TemplateId;
 
 /**
  * Holds the error information from Validators.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Detail implements Serializable {
 	private static final long serialVersionUID = 8818544157552590676L;
+
+	public static Detail forErrorCodeAndNode(LocalizedError code, Node node) {
+		Objects.requireNonNull(node, "node");
+
+		Detail detail = forErrorCode(code);
+		detail.setPath(node.getPath());
+		detail.setTemplateId(node.getType());
+		return detail;
+	}
+
+	public static Detail forErrorCode(LocalizedError code) {
+		Objects.requireNonNull(code, "code");
+
+		Detail detail = new Detail();
+		detail.setErrorCode(code.getErrorCode());
+		detail.setMessage(code.getMessage());
+		return detail;
+	}
+
+	public static void formatMessage(Detail detail, Object... args) {
+		detail.setMessage(String.format(detail.getMessage(), args));
+	}
+
+	@JsonProperty("errorCode")
+	private ErrorCode errorCode;
```

Comment:
I agree. I actually missed this and this was the whole point of the story.... But yes I think an explicit error code would be better.

Reply:
Yes, the request is that each error have a unique code to go along with it so that users can reference the codes.  eg: "Error 1001 - Not a valid XML file" and so forth.

Reply:
So the `ErrorCode` enumeration could just be called `ConversionError` or something and along with the message, each enumerated value would also have a code to go with it.  It should serialize out with the error output as a separate field.

Reply:
done